File: Chapter 02/C#/Section_2_7_1_Weighted_Decision_Model/Option.cs
using System.Collections.Generic;

namespace Chapter02.WeightedDecisionModel;

/// <summary>
/// Represents a single architectural choice to be evaluated.
/// This is a simple data-holding class (POCO).
/// </summary>
public class Option
{
    /// <summary>
    /// The name of the architectural option (e.g., "Redis", "In-Memory Cache").
    /// </summary>
    public required string Name { get; set; }

    /// <summary>
    /// A dictionary holding the scores for this option against various criteria.
    /// The key is the criterion name (e.g., "performance", "cost") and the value
    /// is the score, typically on a scale (e.g., 1 to 5).
    /// </summary>
    public required Dictionary<string, int> Scores { get; set; }
}

File: Chapter 02/Python/section_2_7_1_weighted_decision_model/option.py
from dataclasses import dataclass
from typing import Dict

@dataclass
class Option:
    """
    Represents a single architectural choice to be evaluated.
    
    This is a data class, a simple object whose main purpose is to hold data.
    The `@dataclass` decorator automatically generates special methods like
    `__init__`, making the code more concise.
    """
    name: str
    scores: Dict[str, int]  # e.g., {"performance": 5, "cost": 1}